src/main/java/io/reactivex/Flowable.java

```diff
@@ -2965,6 +2965,19 @@ public static int bufferSize() {
      *  backpressure; if violated, the operator <em>may</em> signal {@code MissingBackpressureException}.</dd>
      *  <dt><b>Scheduler:</b></dt>
      *  <dd>{@code merge} does not operate by default on a particular {@link Scheduler}.</dd>
+     *  <dt><b>Error handling:</b></dt>
+     *  <dd>If any of the source {@code Publisher}s signal a {@code Throwable} via {@code onError}, the resulting
+     *  {@code Flowable} terminates with that {@code Throwable} and all other source {@code Publisher}s are cancelled.
+     *  If more than one {@code Publisher} signals an error, the resulting {@code Flowable} may terminate with the
+     *  first one's error or, depending on the concurrency of the sources, may terminate with a
+     *  {@code CompositeException} containing two or more of the various error signals.
+     *  {@code Throwable}s that didn't make into the composite will be sent (individually) to the global error handler via
+     *  {@link RxJavaPlugins#onError(Throwable)} method as <em>undeliverable errors</em>. Similarly, {@code Throwable}s
+     *  signaled by source(s) after the returned {@code Flowable} has been cancelled or terminated with a
+     *  (composite) error will be sent to the same global error handler.
+     *  Use {@link #mergeDelayError(Iterable)} to merge sources and terminate only when all source {@code Publisher}s
+     *  have completed or failed with an error.
+     *  </dd>
```

Comment:
Does it make sense to explicitly state that split-merge pattern like
```java
PublishSubject<Boolean> ps = PublishSubject.create();

Observable.merge(
        ps.filter((condition) -> condition),
        ps.filter((condition) -> !condition)
).subscribe(
        (next) -> { },
        (error) -> error.printStackTrace()
);

ps.onError(new RuntimeException("Will cause UndeliverableException"));
```
is guaranteed to produce an UndeliverableException in addition to onError?

Maybe in Wiki instead of javadoc?

Reply:
The main effect is the merge here, not the split. Also I'm not sure how often people find out about this behavior through this split-merge pattern.

Reply:
Indeed merge is the main reason, however every time people use merge-split and **don't find out** about this behavior they introduce a potential hard-to-find bug.

It still seems to me that `merge` should by default behave like `mergeFirstErrorOnly` from #5779 discussion, but I'm definitely not pushing.